app/models/ext_management_system.rb

```diff
@@ -186,6 +205,49 @@ def hostname_format_valid?
 
   default_value_for :enabled, true
 
+  after_save :change_maintenance_for_child_managers, :if => proc { |ems| ems.changed_attributes.include?('enabled') }
+
+  def disable!
+    _log.info("Disabling EMS [#{name}] id [#{id}].")
+    update!(:enabled => false)
+    _log.info("Disabling EMS [#{name}] id [#{id}] successful.")
+  end
+
+  def enable!
+    _log.info("Enabling EMS [#{name}] id [#{id}].")
+    update!(:enabled => true)
+    _log.info("Enabling EMS [#{name}] id [#{id}] successful.")
+  end
+
+  # Move ems to maintenance zone and backup current one
+  def pause!
+    _log.info("Pausing EMS [#{name}] id [#{id}].")
+    update!(
+      :zone_before_pause => zone,
+      :zone              => Zone.maintenance_zone,
+      :enabled           => false
+    )
+    _log.info("Pausing EMS [#{name}] id [#{id}] successful.")
+  end
+
+  # Move ems to original zone, reschedule task/jobs/.. collected during maintenance
+  def resume!
+    _log.info("Resuming EMS [#{name}] id [#{id}].")
+
+    new_zone = if zone_before_pause.nil?
+                 zone.visible? ? zone : Zone.default_zone
```

Comment:
Is it possible to get here?

Reply:
actually this is your previous change request. It's possible in existing appliances, because OpenShift allows disabling EMS without moving to maintenance zone now. This PR replaces it, so this code fixes potential inconsistency

Reply:
@jrafanie 
Visibility is term used for Zone. Zone can be visible/invisible.
Disabling is used for ExtManagementSystem. Disabled EMS (EMS.enabled == false) kills it's refresh worker etc.

This PR combines both - disable EMS and assign it to maintenance zone (which is invisible by default)